docs/listings/example_workflows/workflow_with_cache.yml

```diff
@@ -0,0 +1,21 @@
+name: My workflow
+on:
+  push:
+  pull_request:
+jobs:
+  check:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Setup Starknet Foundry
+        id: setup-snfoundry
+        uses: foundry-rs/setup-snfoundry@v3
+      - name: Check versions
+        run: |
+          snforge --version
+          sncast --version
+      - name: Cache Starknet Foundry installation
+        uses: actions/cache@v4
+        with:
+          path: ${{ steps.setup-snfoundry.outputs.starknet-foundry-prefix }}
+          key: ${{ runner.os }}-snfoundry-${{ steps.setup-snfoundry.outputs.starknet-foundry-version }}
```

Comment:
Shouldn't "Cache Starknet Foundry installation" be executed first, before setup-snfoundry? And shouldn't `setup-snfoundry` only trigger on cache hit false?

Reply:
Right, I've added a step which tries to restore from cache.